R/functions.R

```diff
@@ -242,6 +242,34 @@ get_biomart <- function(count_df, synid, version, host, filters, organism) {
     return(biomart_results)
   }
 }
+
+
+#'Check for SampleID Inconsistencies in Counts Matrix and Metadata
+#'
+#'Subsequent steps assume unique SampleIDs are present within both the counts matrix
+#'and the metadata. This function checks for duplicated SampleIDs in both the counts
+#'matrix and metadata before then ensuring each sample in the counts matrix has a 
+#'corresponding line in the metadata, and vice versa. 
+#'@inheritParams get_biomart
+#'@param md A data frame with sample identifiers as row names and relevant experimental covariates.
+#'@export
+check_mismatch <- function(md, count_df) {
+  # Isolate the SampleID's from the metadata and counts matrix
+  md_sampleid <- rownames(md)
+  cnt_sampleid <- colnames(count_df)
+  
+  # First check if these are duplicates in either of these. Returns an error if duplicates are found.
+  if (length(unique(md_sampleid)) != length(md_sampleid) ){
```

Comment:
We can modify this conditional since we inherently know duplicate row names are not allowed. Instead of checking for duplicates, let's start right away with checking the output of `setdiff` as you have it on line 268.